devel/githubContrib.go

```diff
@@ -0,0 +1,239 @@
+		log.Fatal("Need to have GITHUB_TOKEN set in the env")
+	}
+	return token
+}
+
+// GitHubAPIURL returns the full v3 rest api for a given path.
+func GitHubAPIURL(path string) string {
+	return "https://api.github.com/" + path
+}
+
+// NewGhRequest makes a GitHub request (with Accept and Authorization headers).
+func NewGhRequest(url string) *http.Request {
+	req, err := http.NewRequest("GET", url, nil)
+	checkOrDie(err, "Unable to make request")
+	req.Header.Add("Accept", "application/vnd.github.v3+json")
+	req.Header.Add("Authorization", "token "+tokenFromEnv())
+	req.Header.Add("User-Agent", "githubContribExtractor")
+	return req
+}
+
+// GetBodyForURL gets the body or dies/abort on any error.
+func GetBodyForURL(url string) []byte {
+	req := NewGhRequest(url)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	checkOrDie(err, "Unable to send request")
+	body, err := ioutil.ReadAll(resp.Body)
+	checkOrDie(err, "Unable to read response")
+	succ := resp.StatusCode
+	log.Printf("Got %d : %s for %s", succ, resp.Status, url)
+	if succ != http.StatusOK {
+		os.Exit(1)
+	}
+	if *debugFlag {
+		PrettyPrintJSON(body)
+	}
+	return body
+}
+
+// ExtractResult gets the body as json, parses it or dies/abort on any error.
+func ExtractResult(url string, result interface{}) {
+	body := GetBodyForURL(url)
+	err := json.Unmarshal(body, &result)
+	checkOrDie(err, "Unable to parse json")
+}
+
+// PrettyPrintJSON outputs indented version of the Json body (debug only).
+func PrettyPrintJSON(body []byte) {
+	var out bytes.Buffer
+	err := json.Indent(&out, body, "", "  ")
+	checkOrDie(err, "Unable to Indent json")
+	_, err = out.WriteTo(os.Stdout)
+	checkOrDie(err, "Unable to Write json")
+}
+
+// repo is what we use from github rest api v3 listing repositories per org.
+type repo struct {
+	ID              int64  `json:"id"`
+	Name            string `json:"name"`
+	FullName        string `json:"full_name"`
+	ContributorsURL string `json:"contributors_url"`
+	IsFork          bool   `json:"fork"`
+}
+
+// userC is what we care about from what we get from the ContributorsURL.
+type userC struct {
+	Login         string `json:"login"`
+	ID            int64  `json:"id"`
+	Contributions int64  `json:"contributions"`
+}
+
+// userData is for the json we get from the /users/:username API call.
+type userData struct {
+	Login   string `json:"login"`
+	Name    string `json:"name"` // full name
+	Company string `json:"company"`
+	Email   string `json:"email"`
+}
+
+var fromEmailCount = 0 // global variable ftl (or ftw)
+
+// Company returns its best guess of the company for a given GitHub user login.
+func Company(login string, contribCount int64, user *userData) string {
+	ExtractResult(GitHubAPIURL("users/"+login), user)
+	return companyFromUser(*user, contribCount)
+}
+func companyFromUser(user userData, contribCount int64) string {
+	var company string
+	// Strip trailing inc and .com or leading @:
+	r := regexp.MustCompile("[., ]+(com|inc)[ ,.]*$| |^@")
```

Comment:
this is used in a loop, correct?  why not have this be done in init()?

```
var companyRegex *regexp.Regexp
func init() {
	companyRegex = regexp.MustCompile("[., ]+(com|inc)[ ,.]*$| |^@")
}
```

From the regexp docs:
> MustCompile is like Compile but panics if the expression cannot be parsed. It simplifies safe initialization of global variables holding compiled regular expressions.